sdk/servicefabricmanagedclusters/Microsoft.Azure.Management.ServiceFabricManagedClusters/tests/Tests/ApplicationTests.cs

```diff
@@ -0,0 +1,208 @@
+
+namespace ServiceFabricManagedClusters.Tests
+{
+    using System;
+    using Microsoft.Azure.Management.Resources;
+    using Microsoft.Azure.Management.ServiceFabricManagedClusters;
+    using Microsoft.Azure.Management.ServiceFabricManagedClusters.Models;
+    using Microsoft.Rest.ClientRuntime.Azure.TestFramework;
+    using Xunit;
+
+    public class ApplicationTests : ServiceFabricManagedTestBase
+    {
+        internal const string Location = "South Central US";
+        internal const string ResourceGroupPrefix = "sfmc-net-sdk-app-rg-";
+        internal const string ClusterNamePrefix = "sfmcnetsdk";
+
+        private const string AppName = "Voting";
+        private const string AppTypeVersionName = "1.0.0";
+        private const string AppTypeName = "VotingType";
+        private const string AppPackageUrl = "https://sfmconeboxst.blob.core.windows.net/managed-application-deployment/Voting.sfpkg";
+        private const string StatefulServiceName = "VotingData";
+        private const string StatefulServiceTypeName = "VotingDataType";
+
+        [Fact]
+        public void AppCreateDeleteTest()
+        {
+            using (MockContext context = MockContext.Start(this.GetType()))
+            {
+                var serviceFabricMcClient = GetServiceFabricMcClient(context);
+                var resourceClient = GetResourceManagementClient(context);
+                var resourceGroupName = TestUtilities.GenerateName(ResourceGroupPrefix);
+                var clusterName = TestUtilities.GenerateName(ClusterNamePrefix);
+                var nodeTypeName = TestUtilities.GenerateName("nt");
+
+                var cluster = this.CreateManagedCluster(resourceClient, serviceFabricMcClient, resourceGroupName, Location, clusterName, sku: "Basic");
+                cluster = serviceFabricMcClient.ManagedClusters.Get(resourceGroupName, clusterName);
+                Assert.NotNull(cluster);
+                Assert.Equal("Succeeded", cluster.ProvisioningState);
+                Assert.Equal("WaitingForNodes", cluster.ClusterState);
+                var primaryNodeType = this.CreateNodeType(serviceFabricMcClient, resourceGroupName, clusterName, nodeTypeName, isPrimary: true, vmInstanceCount: 6);
+
+                this.WaitForClusterReadyState(serviceFabricMcClient, resourceGroupName, clusterName);
+
+                CreateAppType(serviceFabricMcClient, resourceGroupName, clusterName, cluster.ClusterId, AppTypeName);
+                var appTypeVersion = CreateAppTypeVersion(serviceFabricMcClient, resourceGroupName, clusterName, cluster.ClusterId, AppTypeName, AppTypeVersionName, AppPackageUrl);
+                CreateApplication(serviceFabricMcClient, resourceGroupName, clusterName, cluster.ClusterId, AppName, appTypeVersion.Id);
+                CreateService(serviceFabricMcClient, resourceGroupName, clusterName, cluster.ClusterId, AppName, StatefulServiceTypeName, StatefulServiceName);
```

Comment:
There are no assertions over the properties of the returned types.  It is generally a good idea to ensure that the serialization/deserialization of properties is working end-to-end in at east one test

Reply:
thanks, I added properties validations